Akka.Serialization.MessagePack/MessagePackTypeFilteringOptions.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using MessagePack;
+using MessagePack.Formatters;
+
+namespace Akka.Serialization.MessagePack
+{
+    public class MessagePackTypeFilteringOptions : MessagePackSerializerOptions
```

Comment:
This is the same safety mechanism we added for Hyperion, right? Is it also end-user extensible the way Hyperion's is?

Reply:
This impl doesn't have the end user extensibility, I stole it from my original Hyperion PR.

That said, I would assume that also means it should be easy to add the extensible filtering. :)